-Changes to enable docker container to run

kcammel opened the pull request:
-Changed Dockerfile to include wrapper script
   -Changed config.yaml to work with docker mounted
-Changed shutdown handling
  -Removes symlinks
  -Creates plot for most recent iteration and places them in main output dir
  -Clean shutdown handling (waiting for processes to end)

kcammel committed:
-Changes to enable docker container to run
   -Changed Dockerfile to include wrapper script
   -Changed config.yaml to work with docker mounted
-Changed shutdown handling
  -Removes symlinks
  -Creates plot for most recent iteration and places them in main output dir
  -Clean shutdown handling (waiting for processes to end)

AlexJanse commented on Dockerfile:
```diff
@@ -1,83 +1,57 @@
+ARG MODKIT_VERSION=v0.5.0
+
 # Base image to compile R libraries
 ARG R_LIB_COMPILER_IMAGE=princessmaximacenter/debian_r:4.4.1
```
Maybe good to double check which variable are not used any more and remove them if they don't provide any additional information

AlexJanse commented on Dockerfile:
```diff
@@ -1,83 +1,57 @@
-ARG BASE_PYTHON_IMAGE_DIGEST=sha256:dfc6d3e53d5ecbb6a5de856dafa9c85be5ce0fbed3cb005f3382d3f8f0ac07da
-ARG IMAGE_NAME=sturgeon
-ARG IMAGE_VERSION=1.0.0
+ARG BASE_PYTHON_IMAGE_VERSION=3.9.6-slim-buster
```
Buster is end-of-life so for the next version we need to consider moving to bookworm or trixie. For now if it builds it is fine

AlexJanse commented on Dockerfile:
```diff
@@ -86,23 +60,20 @@ RUN apt update \
     && rm -rf /var/lib/apt/lists/*
 
-RUN find / -name "*nvidia*" -exec rm -rf {} \; 2>/dev/null
-
```
Given that we gonna share this container with the world it is nice to add some labels to know where it is from and what is in it. The following is an example of one of our containers:
```
LABEL org.opencontainers.image.title="${IMAGE_NAME}" \
      org.opencontainers.image.version="${IMAGE_VERSION}" \
      org.opencontainers.image.authors="PMC Genomicscore" \
      org.opencontainers.image.description="R ${IMAGE_VERSION} image with library compiler" \
      org.opencontainers.image.source="https://bitbucket.org/princessmaximacenter/wdl_pipeline/src/dev/" \
      org.opencontainers.image.base.name="${BASE_IMAGE_NAME}:${BASE_IMAGE_VERSION}"\
      org.opencontainers.image.base.digest="${BASE_IMAGE_DIGEST}" \
      org.opencontainers.image.url="https://bitbucket.org/princessmaximacenter/wdl_pipeline/src/dev/" \
      apps.r.source="http://cloud.r-project.org/bin/linux/debian" \
      apps.r.version="${R_VERSION}" 
```
      
I haven't made these labels myself so you can maybe look around what labels are possible. Good to somehow mention either the PMC/UMC and/or link to the git repos. Not sure how extensive it needs to be but maybe you can look for examples of other containers outside our own.

AlexJanse commented:
Given people outside our group are going to look at it, it might be good to go through the README and adjust where needed.

kcammel committed:
Update README for easier use

kcammel committed:
Updated README to include docker run
Changed Dockerfile to include labels and image version for sturgeon
Changed config.yaml to work with docker installation (without needing to have installed sturgeon locally_

kcammel committed:
Edited Dockerfile and R script to deal with buster being end-of-life.
Needed to update bioconductor to v3.21, removed rpy2 from pyproject.toml

Updated readme for running the docker image correctly
